core/src/test/java/com/backblaze/b2/json/B2JsonTest.java

```diff
@@ -1801,4 +1837,77 @@ private SubclassH(String name) {
         }
     }
 
+    @Test
+    public void testRequiredFieldNotInVersion() throws B2JsonException {
+        final String json = "{}";
+        final B2JsonOptions options = B2JsonOptions.builder().setVersion(1).build();
+        final VersionedContainer obj = bzJson.fromJson(json, VersionedContainer.class, options);
+        assertEquals(0, obj.x);
+        assertEquals(1, obj.version);
+    }
+
+    @Test
+    public void testRequiredFieldMissingInVersion() throws B2JsonException {
+        final String json = "{}";
+        final B2JsonOptions options = B2JsonOptions.builder().setVersion(5).build();
+
+        thrown.expectMessage("required field x is missing");
+        bzJson.fromJson(json, VersionedContainer.class, options);
+    }
+
+    @Test
+    public void testFieldPresentButNotInVersion() throws B2JsonException {
+        final String json = "{ \"x\": 5 }";
+        final B2JsonOptions options = B2JsonOptions.builder().setVersion(1).build();
+
+        thrown.expectMessage("field x is not in version 1");
+        bzJson.fromJson(json, VersionedContainer.class, options);
+    }
+
+    @Test
+    public void testFieldPresentAndInVersion() throws B2JsonException {
+        final String json = "{ \"x\": 5 }";
+        final B2JsonOptions options = B2JsonOptions.builder().setVersion(10).build();
+        final VersionedContainer obj = bzJson.fromJson(json, VersionedContainer.class, options);
+        assertEquals(5, obj.x);
+        assertEquals(10, obj.version);
+    }
+
+    @Test
+    public void testSerializeSkipFieldNotInVersion() throws B2JsonException {
+        final B2JsonOptions options = B2JsonOptions.builder().setVersion(1).build();
```

Comment:
would setVersion(3) be better?  :)

Reply:
sorry if i just haven't seen it yet or if it hasn't sunk in.  are there any tests for a field going away (to test lastVersion?).

Reply:
will add